# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1512591724 21600
#      Wed Dec 06 14:22:04 2017 -0600
# Branch stable
# Node ID d8626fd6bb3692cf03864c1d5e14ceee210807de
# Parent  b71ba49473f4f749c8b92cd6c10b7ef2a1fa3ff2
For gnuplot toolkit, do not map TrueColor data to colormap size (bug #52599).

* __gnuplot_draw_axes__.m (mapcdata): Use 'cdata' as a return variable rather
  than 'retval'.  Don't scale the TrueColor data to range cmap_sz.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -2863,30 +2863,28 @@ function maybe_do_x2tick_mirror (plot_st
 
   if (! isempty(axis_obj.xtick))
     fprintf (plot_stream, "unset xtics; set x2tics %s nomirror\n",
                           axis_obj.tickdir);
   endif
 
 endfunction
 
-function retval = mapcdata (cdata, mode, clim, cmap_sz)
+function cdata = mapcdata (cdata, mode, clim, cmap_sz)
   if (ndims (cdata) == 3)
     ## True Color, clamp data to 8-bit
     clim = double (clim);
     cdata = double (cdata);
     clim_rng = clim(2) - clim(1);
     if (clim_rng != 0)
       cdata = 255 * (cdata - clim(1)) / clim_rng;
       cdata(cdata < 0) = 0;  cdata(cdata > 255) = 255;
     else
       cdata(:) = 255;
     endif
-    ## Scale using inverse of gnuplot's cbrange mapping
-    retval = 1 + cdata * (cmap_sz-1)/255;
   else
     if (islogical (cdata))
       cdata += 1;
     elseif (strcmp (mode, "scaled"))
       clim = double (clim);
       cdata = double (cdata);
       clim_rng = clim(2) - clim(1);
       if (clim_rng != 0)
@@ -2896,11 +2894,11 @@ function retval = mapcdata (cdata, mode,
       endif
     else
       if (isinteger (cdata))
         cdata += 1;
       else
         cdata = fix (cdata);
       endif
     endif
-    retval = max (1, min (cdata, cmap_sz));
+    cdata = max (1, min (cdata, cmap_sz));
   endif
 endfunction
